pkg/hostagent/requirements.go

```diff
@@ -170,6 +170,19 @@ Also see "/var/log/cloud-init-output.log" in the guest.
 `,
 			})
 	}
+	if *a.y.SSH.ForwardAgent {
+		req = append(req, requirement{
```

Comment:
The `requirement` script are designed for validation, not for mutation of the VM.
So this should be rather moved to the (custom) provision script.
(And then we don't need to care about the file path, as it is customizable in the script?)

Reply:
I'm not sure where that is but I will look. I'm hoping to make this improvement for all users of forwardAgent so they don't have to do the hack mentioned in the issue. Similar to Docker Desktop where the agent sock was in a static location. Was one issue we ran into when migrating from Docker Desktop.

Reply:
> So this should be rather moved to the (custom) provision script.

Yes, but it would require something like #287 to be implemented first to run the script after the requirements scripts are done. The current provisioning scripts run before the host agent has established the ssh connection, so `SSH_AUTH_SOCK` is not yet available.

Reply:
> Similar to Docker Desktop where the agent sock was in a static location.

I didn't realize Docker Desktop did this. It uses `/run/host-services/ssh-auth`, so I would propose to use the same location for lima, if we are going to agree on a way to implement this.

Reply:
Using `/run/host-services/ssh-auth` is ok for me, but this has to be moved to the provision script or the boot script, as the `requirements` are designed for validation scripts.

Reply:
I understand, but it means we need to implement an additional type of provisioning script that will be executed by the host agent once the SSH connection has been setup (which is essentially #287, except that one also asks for the sshfs mounts to be done).

I'll leave it to @ryancurrah to decide if he wants to help with implementing this. Otherwise I'm fine with implementing it just for Rancher Desktop for now (simply by running `lima ln ...` once the VM has started, which will be a trivial change).

Reply:
> this has to be moved to the provision script or the boot script

@AkihiroSuda Are you opposed to implementing is directly in the hostagent code, either before or after setting up the sshfs mounts? That would be a lot simpler, as it doesn't need to implement a new type of provisioning script first.

Reply:
Hostagent is ok, but not in requirements.go

Reply:
Ack I will look at doing this in hostagent code.

Reply:
Started looking at this. So if we are trying to be compatible with Docker ssh-agent this is how it is laid out on the Docker VM.

```bash
╰─ docker run -it --rm --privileged --pid=host justincormack/nsenter1
bash-5.0# ls -l /run | grep host-
drwxr-xr-x    2 root     root           300 Jan 30 20:00 host-services

bash-5.0# ls -l /run/host-services/
srwxr-xr-x    1 root     root             0 Jan 30 20:00 backend.native.sock
srwxr-xr-x    1 root     root             0 Jan 30 20:00 backend.sock
srwxr-xr-x    1 root     root             0 Jan 30 20:00 dns.internal.grpc.sock
srwxr-xr-x    1 root     root             0 Jan 30 20:00 dns.system.grpc.sock
srwxr-xr-x    1 root     root             0 Jan 30 20:00 docker.proxy.sock
srwxr-xr-x    1 root     root             0 Jan 30 20:00 extension-manager.sock
srwxr-xr-x    1 root     root             0 Jan 30 20:00 filesystem.sock
srwxr-xr-x    1 root     root             0 Jan 30 20:00 hubproxy.sock
srwxr-xr-x    1 root     root             0 Jan 30 20:00 ntp.udp.sock
srwxr-xr-x    1 root     root             0 Jan 30 20:00 socks.sock
srwxr-xr-x    1 root     root             0 Jan 30 20:00 ssh-auth.sock
srwxr-xr-x    1 root     root             0 Jan 30 20:00 vpnkit-data.sock
srwxr-xr-x    1 root     root             0 Jan 30 20:00 vpnkit.sock
```

Reply:
To create a folder and link in `/run` I will need to use sudo.

Reply:
So my changes work when tested locally, but not sure if this is ideal. 

Three concerns from me right now:
- Having to use `sudo` to create the directory.
- The openness of the permissions to read from the socket.
- The area within `hostagent.go` where I put the code.

The verification steps are below.

Verify socket got created on Lima VM.

```
╰─ _output/bin/limactl shell default

ryancurrah@lima-default:/Users/ryancurrah/git/github.com/ryancurrah/lima$ ls -l /run/ | grep host-
drwxr-xr-x  2 root root   60 Jan 31 02:28 host-services

ryancurrah@lima-default:/Users/ryancurrah/git/github.com/ryancurrah/lima$ ls -l /run/host-services/
lrwxrwxrwx 1 root root 30 Jan 31 02:28 ssh-auth.sock -> /tmp/ssh-XXXX5CLqMY/agent.2334
```

Verify ssh-agent works in a container.

```
╰─ lima nerdctl run -it -v /run/host-services/ssh-auth.sock:/run/host-services/ssh-auth.sock -e SSH_AUTH_SOCK=/run/host-services/ssh-auth.sock golang:latest bash

root@03723ef10743:/go# ssh-add -l
4096 SHA256:......KOd0ivA ryan@currah.ca (RSA)
```

Reply:
> * Having to use `sudo` to create the directory and link.

I think this is fine.

> * The openness of the permissions to read from the socket.

These are the permissions on the symlink. The user still needs to satisfy the permissions of the symlink target. So I think this is fine too.

> * The area within `hostagent.go` where I put the code.

I think it should go directly before setting up the `sshfs` mounts.

Reply:
> These are the permissions on the symlink. 

Well, they will be in our case. For Docker Desktop these seem to be the sockets directly. But they probably don't care because unlike Lima they don't expect users to run **anything** directly on the VM, but only in the containers.

Reply:
Moved the code and confirmed it still works. I think I moved it to the right place.